### PR TITLE
upgrade version of amplify to v5

### PR DESCRIPTION
**Pull request by katieklein**
*Issue #, if available:*
Stepping the learn site up to aws-amplify v5

staging: https://amplify-upgrade.dm6i4p3zktlna.amplifyapp.com/learn

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -37,11 +37,14 @@ env.local
 
 amplify/team-provider-info.json
 
+credentials.csv
```
Comment (404ghosting) [inline .gitignore:40]: Place in alphabetical order please

File: .gitignore [hunk 2]
```diff
@@ -37,11 +37,14 @@ env.local
...
+
 #amplify-do-not-edit-begin
 amplify/\#current-cloud-backend
 amplify/.config/local-*
 amplify/logs
 amplify/mock-data
+amplify/mock-api-resources
```
Comment (404ghosting) [inline .gitignore:47]: Same feed back as above

**Review by ghost — CHANGES_REQUESTED** [review]
Blocking since there is no staging link

File: .gitignore (hunk 1 above)
Comment (404ghosting) [reply .gitignore:40]: @reesscot what do you think?

**Review by ghost — COMMENTED** [review]
Please add description of your changes in your pr

**Comment by jacoblogan** [comment]
Not sure if this staging site is up to date https://amplify-upgrade.dm6i4p3zktlna.amplifyapp.com/ but it looks like we might be losing some styling or theming with the upgrade
Screenshot 2024-06-27 at 4 21 32 PM

**Comment by 404ghosting** [comment]
No courses found!!

**Review by jacoblogan — APPROVED** [review]

File: src/lib/getData.ts [hunk 3]
```diff
@@ -144,8 +155,17 @@ export async function getCourseContributors(
...
   const contributorCourses = await DataStore.query(ContributorCourse);
+  const resolvedCourses = [];
+
+  for (const contributorCourse of contributorCourses) {
+    const id = await contributorCourse.id;
+    const contributor = await contributorCourse.contributor;
+
+    const course = await contributorCourse.course;
+    resolvedCourses.push({id, contributor:contributor, course: course}); 
+  }
```
Comment (timngyn) [inline src/lib/getData.ts:166]: I'm a little confused by this loop, why do we have it? It looks like it's just accessing the properties from `contributorCourses` and just making a new object array to filter? Is there a reason we need to `await` the properties too?

File: src/models/index.d.ts
Comment (timngyn) [inline src/models/index.d.ts:1]: Was this updated by Amplify? Would you happen to know why it has the `Eager` and `Lazy` word in front of the type names?

**Pull request closed without merging**